kube-runtime/src/reflector/store.rs

```diff
@@ -105,39 +108,66 @@ where
                 let key = obj.to_object_ref(self.dyntype.clone());
                 self.store.write().remove(&key);
             }
-            watcher::Event::Restarted(new_objs) => {
+            watcher::Event::RestartedStart => {
+                let mut buffer = self.buffer.write();
+                *buffer = AHashMap::new();
```

Comment:
`self.buffer.write().clear()` might work as a shorter variant

Reply:
`clear()` keeps the allocated memory for reuse (https://docs.rs/ahash/latest/ahash/struct.AHashMap.html#method.clear). By re-assigning the buffer to a new `AHashMap` the memory will be deallocated, because assignment drops the left-hand side

Reply:
ah, that might be worth putting in the comment so we don't accidentally break that in a refactor later. we don't have any benches atm.